### DISCUSSION TITLE
Reconsider the old import syntax

### DISCUSSION THREAD

**Discussion by bohonghuang** [scala/scala3#17876] [Feature Requests]
The new import syntax looks familiar and is friendly to newcomer from other language, but it seems not so suitable for Scala. 

The import syntax should be distinct from the legitimate identifier as possible to make it clear. In Scala, both `*` and `as` are legitimate identifiers while in other language such as Rust and Python they are not.  `` ` ``  can be used for disambiguation, but it should not be encouraged outside of test code, and the new import syntax forces the usage of `` ` `` when just importing `*`, which is useful in some calculation-specific codes.

The soft keyword `as` was dropped in given syntax and pattern binding and is only used in import syntax currently, which seems to violate the design philosophy of Scala.

The old import syntax looks perfect, since neither `_` nor `=>` is treated as a legitimate identifier so they can be easily recognized in the long import list. So the old import syntax may be a better choice than the new one?

**Comment by som-snytt** [comment]
I suggest we "normalize" backticks. I prefer it for test method names which are never referenced in code.
```
def `test ticket 1001 which claims we can't have nice things` = ???
```

I remember I contributed something about backticked asterisk (and say that ten times fast) to Scala 2. Was that for patterns or imports? Maybe it was regular asterisk but backticked underscore, since Scala 2 supports underscore as a name for (misconceived) reasons of compatibility.

In any case, this ship of syntax has sailed.

Another angle is that `*` is regex syntax and is implicated in repeating param syntax. That is, there is always some impedance mismatch.

Although I liked "underscore everywhere" in principle, in practice the new syntax is tractable and doesn't generate tickets.
